stable/pomerium/templates/proxy-deployment.yaml

```diff
@@ -0,0 +1,146 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{ template "pomerium.name" . }}
+    chart: {{ template "pomerium.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: {{ template "pomerium.fullname" . }}-proxy
+{{- if .Values.annotations }}
+  annotations:
+{{ toYaml .Values.annotations | indent 4 }}
+{{- end }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "pomerium.name" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+{{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+      labels:
+        app: {{ template "pomerium.name" . }}
+        chart: {{ template "pomerium.chart" . }}
+        heritage: {{ .Release.Service }}
+        release: {{ .Release.Name }}
+{{- if .Values.podLabels }}
+{{ toYaml .Values.podLabels | indent 8 }}
+{{- end }}
+    spec:
+{{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName }}
+{{- end }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args:
+{{- range $key, $value := .Values.extraArgs }}
+{{- if $value }}
+          - --{{ $key }}={{ $value }}
+{{- else }}
+          - --{{ $key }}
+{{- end }}
+{{- end }}
+        env:
+        - name: SERVICES
+          value: authorize
+        - name: COOKIE_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "pomerium.fullname" . }}
+              key: cookie-secret
+        - name: SHARED_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "pomerium.fullname" . }}
+              key: shared-secret             
+        - name: AUTHENTICATE_SERVICE_URL
+          value: {{ .Values.proxy.authenticateServiceUrl }}
+        - name: AUTHORIZE_SERVICE_URL
+          value: {{ .Values.proxy.authorizeServiceUrl }}
```

Comment:
@desimone it could be interesting for the chart to directly rely on  `AUTHENTICATE_INTERNAL_URL` and `AUTHORIZE_INTERNAL_URL` since they are deployed in the same cluster, but maybe it is on purpose? I remember having problem with that with buzzfeed-sso behind a nginx-ingress…

Reply:
I've added some code to infer the internal url if none is provided. Behind the ingress communication is especially tricky, and is basically requires the use of `AUTHENTICATE_INTERNAL_URL` and `AUTHORIZE_INTERNAL_URL` in kubernetes and docker-compose environments.